react/modules/useInterpolatedLink.ts

```diff
@@ -1,37 +1,36 @@
 import { useRuntime } from 'vtex.render-runtime'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
+import type { ProductTypes } from 'vtex.product-context'
+import type { RenderContext } from 'vtex.render-runtime'
 
 import interpolateLink from './interpolateLink'
 import { AvailableContext } from './mappings'
 
-interface Context {
+export interface Context {
   type: AvailableContext
   namespace?: string
-  context: Record<string, unknown>
+  context?: Partial<ProductTypes.ProductContextState> | Record<string, any>
```

Comment:
Maybe this would be more accurate

```suggestion
  context: Partial<ProductTypes.ProductContextState> | Record<string, any> | undefined
```